### ISSUE TITLE
Node Module Error (newline-json\parser.js)

### ISSUE THREAD

**Issue by jleelove** [feross/webtorrent-hybrid#44]
It seams like there is a error within one of the dependencies. I'm not sure if this is just a windows problem.
```
events.js:160
      throw er; // Unhandled 'error' event
      ^

Error: Unexpected end of JSON input in "\n\"starting\"\n"
    at Parser._transform (G:\btproject\node_modules\newline-json\parser.js:30:12)
    at Parser.Transform._read (_stream_transform.js:167:10)
    at Parser.Transform._write (_stream_transform.js:155:12)
    at doWrite (_stream_writable.js:307:12)
    at writeOrBuffer (_stream_writable.js:293:5)
    at Parser.Writable.write (_stream_writable.js:220:11)
    at Socket.ondata (_stream_readable.js:556:20)
    at emitOne (events.js:96:13)
    at Socket.emit (events.js:188:7)
    at readableAddChunk (_stream_readable.js:177:18)
```

**Comment by ipsBruno** [comment]
Same error here.

Aparrently, is problem of new-linejson

https://github.com/mappum/electron-eval/issues/50

@mappum @feross

**Comment by mappum** [comment]
It appears both occurrences have happened on Windows. Also, `newline-json` gives that error if the data begins with `\n`, so we just have to find where that newline is coming from.

**Comment by jleelove** [comment]
[parse.js](https://github.com/CrowdProcess/newline-json/blob/master/parser.js#L30-L31)
commenting out lines 30 and 31 are a band-aid let me know if helps you

**Comment by ipsBruno** [comment]
I already try commenting out line. But don't still work here.

**Comment by mappum** [comment]
That `newline-json` stream is receiving the stdout output of the Electron child process. So I think we will just have to ensure we consume the first byte (`\n`) of the stdout stream when using Windows (which sucks).

**Comment by jleelove** [comment]
why is windows the problem?

**Comment by mappum** [comment]
Can you guys test with `electron-eval` on this branch: https://github.com/mappum/electron-eval/tree/newline-fix?

**Comment by ipsBruno** [comment]
Nothing here.

I made all changes. 

https://github.com/mappum/electron-eval/compare/newline-fix

And still show error here.

**Comment by ipsBruno** [comment]
I try with this:

```javascript
    this.child.stdout.once('data', (data) => {
      // this hack fixes issues where some environments (namely Windows)
      // prepend a '\n' at the beginning of the stdout stream. This breaks
      // the JSON parser, so if the stream starts with '\n' we skip it
      if (data[0] === '\n'.charCodeAt(0)) {
        data = data.slice(1)
      }
      //this.stdout.write(data)
      //this.child.stdout.pipe(this.stdout)

    })

```

And is working for now, here in Windows.


Thank you

**Comment by mappum** [comment]
This should be fixed as of `electron-eval@0.9.2`. Can you guys reinstall and confirm that the issue is fixed?

**Comment by feross** [comment]
Just tested on Windows. This looks fixed to me. Thanks @mappum.
